torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -39,6 +39,11 @@ constexpr int64_t kCommInitBusyWaitMillis = 2;
 #define NCCL_REMOTE_ERROR
 #endif
 
+#define HAS_NCCL_MAJOR_MINOR (defined(NCCL_MAJOR) && defined(NCCL_MINOR))
+#if HAS_NCCL_MAJOR_MINOR
+static_assert((NCCL_MAJOR == 2 && NCCL_MINOR >= 4) ||   (NCCL_MAJOR > 2), "NCCL version must be 2.4 or later");                                                            
+#endif  // HAS_NCCL_MAJOR_MINOR
+
```

Comment:
To work around the CI's concern, we can directly static_assert. I should be fair to assume both `NCCL_MAJOR` and `NCCL_MINOR` are defined when nccl.h is included.

Reply:
I was concerned about the case where  someone builds PyTorch but with no NCCL library. 
So this file would unnecessarily assert for this case.
If you are fairly sure that NCCL is always required to build PyTorch, I can add the static_assert unconditionally?